lib/src/api.dart

```diff
@@ -95,35 +96,56 @@ class API {
     }
   }
 
+  Map<String, dynamic> _isMap(dynamic value) {
+    if(value is! Map<String, dynamic>)
+      throw APIException('Unexpected type "${value.runtimeType}". '
+        'Expected: "Map<String, dynamic>".');
+    return value;
+  }
+
+  Iterable<dynamic> _isIterable(dynamic value) {
+    if(value is! Iterable<dynamic>)
+      throw APIException('Unexpected type "${value.runtimeType}". '
+        'Expected: "Iterable<dynamic>".');
+    return value;
+  }
```

Comment:
It is not "is Thing" its is "as thing", also why do you even need this, aren't mappers should handle everything?